core/src/HPX/Kokkos_HPX.hpp

```diff
@@ -82,7 +82,7 @@ class hpx_thread_buffer {
   hpx_thread_buffer &operator=(hpx_thread_buffer)         = delete;
 
   void resize(const std::size_t num_threads, const std::size_t size_per_thread,
-              const std::size_t extra_space = 0) noexcept;
+              const std::size_t extra_space = 0);
```

Comment:
Considering that it is always called from `ParallelX::setup()` that does not have a `noexcept` specifier, I think dropping the `noexcept` here is more appropriate than attempting to handle exception or than suppressing the Clang-Tidy diagnostic.